Zend/zend_compile.c

```diff
@@ -3025,7 +3068,15 @@ void zend_compile_assign(znode *result, zend_ast *ast) /* {{{ */
 				/* list($a, $b) = $a should evaluate the right $a first */
 				zend_compile_simple_var_no_cv(&expr_node, expr_ast, BP_VAR_R, 0);
 			} else {
```

Comment:
The zend_list_has_assign_to_self() branch above currently doesn't handle the reference case, it also needs to switch to BP_VAR_W (+ MAKE_REF).

Reply:
So, fwiw, `zend_compile_simple_var_no_cv` does seem to accept the `BP_` type, but then proceeds to do nothing with that argument.  Each emit in there is hardcoded for an `_R` type as well.

Reply:
Sorry for the delay here. I've just pushed a tweak (https://github.com/php/php-src/commit/9fbb019848cbb0387cb7267489dd91ebb4d9050e) so that `zend_compile_simple_var_no_cv` does not require and explicit `zend_adjust_for_fetch_type` call after it, so just passing the right fetch type should work now.